fedimint-client/src/module/mod.rs

```diff
@@ -279,6 +290,30 @@ where
             .await
     }
 
+    // TODO: rename `module_autocommit` to something else and make this a default
+    pub async fn module_autocommit_2<'s, 'dbtx, F, T>(
+        &'s self,
+        tx_fn: F,
+        max_attempts: Option<usize>,
+    ) -> anyhow::Result<T>
```

Comment:
Why not allow arbitrary errors to be returned from the closure?

Reply:
IIRC I had an infalliable code and type inference wanted an error type, which just annoyed me, combined with having to handle failed commit. I don't think strong about it, but the plan was to have a "easy to use" default `module_autocommit` using anyhow (since we use anyhow almost for everything anyway), and another `module_autocommit_something` for cases when someone really wants to handle custom errors / commit fail.